File: app/API/preloved-api/app/Models/Product.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Product extends Model
{
    protected $primaryKey = 'product_id';

    protected $fillable = [
        'seller_id',
        'category_id',
        'nama_barang',
        'deskripsi',
        'harga_jual',
        'harga_asli',
        'kondisi',
        'merek',
        'warna',
        'lokasi_kota',
        'status_barang',
        'waktu_post'
    ];

    public function images()
{
    return $this->hasMany(
        ProductImage::class,
        'product_id',
        'product_id'
    );
}
    public function seller()
    {
        return $this->belongsTo(User::class,'seller_id');
    }

    public function category()
    {
        return $this->belongsTo(Category::class,'category_id');
    }

    public function favorites()
    {
        return $this->hasMany(Favorite::class,'product_id');
    }

    public function offers()
    {
        return $this->hasMany(Offer::class,'product_id');
    }

    public function orders()
    {
        return $this->hasMany(Order::class,'product_id');
    }

    public function chats()
    {
        return $this->hasMany(Chat::class,'product_id');
    }
}

File: app/src/main/java/com/example/preloved/models/Product.java
package com.example.preloved.models;

import com.google.gson.annotations.SerializedName;
import java.io.Serializable;
import java.util.List;

public class Product implements Serializable {

    // Kita sediakan backup "id" jika Laravel me-return dengan key standar resource
    @SerializedName(value = "product_id", alternate = {"id"})
    private int productId;

    @SerializedName("category_id")
    private int categoryId;

    @SerializedName("seller_id")
    private int sellerId;

    @SerializedName("nama_barang")
    private String nama_barang;

    @SerializedName("deskripsi")
    private String deskripsi;

    @SerializedName("harga_jual")
    private String harga_jual;

    @SerializedName("harga_asli")
    private String harga_asli;

    @SerializedName("kondisi")
    private String kondisi;

    @SerializedName("merek")
    private String merek;

    @SerializedName("warna")
    private String warna;

    // [FIX UTAMA]: Diubah ke "seller" agar sinkron dengan relasi model Laravel lu bray!
    @SerializedName("seller")
    private Seller seller;

    @SerializedName("lokasi_kota")
    private String lokasi_kota;

    @SerializedName("status_barang")
    private String status_barang;

    @SerializedName("images")
    private List<ProductImage> images;

    // Getter methods
    public int getProductId() {
        return productId;
    }

    // Backup method untuk activity lama lu jika ada yang terlanjur memanggil getId()
    public int getId() {
        return productId;
    }

    public int getCategoryId() {
        return categoryId;
    }

    public int getSellerId() {
        return sellerId;
    }

    public String getNama_barang() {
        return nama_barang;
    }

    public String getDeskripsi() {
        return deskripsi;
    }

    public String getHarga_jual() {
        return harga_jual;
    }

    public String getHarga_asli() {
        return harga_asli;
    }

    public String getKondisi() {
        return kondisi;
    }

    public String getMerek() {
        return merek;
    }

    public String getWarna() {
        return warna;
    }

    public String getLokasi_kota() {
        return lokasi_kota;
    }

    public Seller getSeller() {
        return seller;
    }

    public String getStatus_barang() {
        return status_barang;
    }

    public List<ProductImage> getImages() {
        return images;
    }
}
